components/server/ee/src/billing/billing-mode.ts

```diff
@@ -140,26 +145,31 @@ export class BillingModesImpl implements BillingModes {
         const hasUbbPaidTeam = teamsModes.some((tm) => tm.mode === "usage-based" && !!tm.paid);
         const hasCbTeam = teamsModes.some((tm) => tm.mode === "chargebee");
         const hasCbTeamSeat = cbTeamSubscriptions.length > 0;
+        const hasCbTeamSubscription = cbOwnedTeamSubscriptions.length > 0;
 
-        if (hasUbbPaidTeam || hasUbbPersonal) {
-            // UBB is greedy: once a user has at least a paid team membership, they should benefit from it!
+        function usageBased() {
             const result: BillingMode = { mode: "usage-based" };
             if (hasCbTeam) {
                 result.hasChargebeeTeamPlan = true;
             }
-            if (hasCbTeamSeat) {
+            if (hasCbTeamSeat || hasCbTeamSubscription) {
                 result.hasChargebeeTeamSubscription = true;
             }
             return result;
         }
+
+        if (hasUbbPaidTeam || hasUbbPersonal) {
+            // UBB is greedy: once a user has at least a paid team membership, they should benefit from it!
+            return usageBased();
```

Comment:
that all makes sense, and it works. it's just quite hard to read.

Reply:
> it's just quite hard to read.

Yeah, that's the only downside I see here :confused: Still hoping all the complexity here is going to go away with Chargbee. :lotus_position: